src/Raven.Server/Platform/Posix/LimitsReader.cs

```diff
@@ -0,0 +1,279 @@
+﻿using System;
+using System.Buffers;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.IO;
+using System.IO.Pipelines;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Server.Monitoring.Snmp;
+using Sparrow.Platform;
+
+namespace Raven.Server.Platform.Posix
+{
+    public static class LimitsReader
+    {
+        public static readonly string MaxMapCountFilePath = "/proc/sys/vm/max_map_count";
+        public static readonly string CurrentMapCountFilePath = "/proc/self/maps";
+
+        public static readonly string ThreadsMaxFilePath = "/proc/sys/kernel/threads-max";
+        public static readonly string ThreadsCurrentFilePath = "/proc/self/stat";
+
+        private static readonly SemaphoreSlim _lock = new SemaphoreSlim(1, 1);
+
+        public static async Task<LimitsInfo> ReadCurrentLimitsAsync()
+        {
+            if (PlatformDetails.RunningOnPosix == false)
+                throw new InvalidOperationException("Cannot read Current Limits because it requires POSIX");
+
+            if (await _lock.WaitAsync(0) == false)
+                return LimitsInfo.Current;
+
+            try
+            {
+                LimitsInfo.Current.MapCountCurrent = await GetCurrentMapCountAsync();
+                LimitsInfo.Current.ThreadsCurrent = GetCurrentThreadsCount();
+                LimitsInfo.Current.SetValues();
+            }
+            finally
+            {
+                _lock.Release();
+            }
+
+            return LimitsInfo.Current;
+        }
+
+        public static LimitsInfo ReadMaxLimits()
+        {
+            if (PlatformDetails.RunningOnPosix == false)
+                throw new InvalidOperationException("Cannot read Current Limits because it requires POSIX");
+
+            if (_lock.Wait(0) == false)
```

Comment:
IMO you want to wait here infinitely, because this is done once per hour and you do not want to wait another one

Reply:
done